Validation/MuonGEMDigis/src/GEMCheckGeometry.cc

```diff
@@ -0,0 +1,105 @@
+#include "Validation/MuonGEMDigis/interface/GEMCheckGeometry.h"
+#include <iomanip>
+
+GEMCheckGeometry::GEMCheckGeometry(const edm::ParameterSet& gc)
+{
+  
+  GE11PhiBegin_ = gc.getUntrackedParameter< double >("GE11PhiBegin",-5.) ;
+  GE11PhiStep_ = gc.getUntrackedParameter< double >("GE11PhiStep",10) ;
+  minPhi_ = gc.getUntrackedParameter< double >("minPhi",-180.);
+  maxPhi_ = gc.getUntrackedParameter< double >("maxPhi",+180.);
+  detailPlot_ = gc.getParameter< bool >("detailPlot"); 
+}
+
+
+void GEMCheckGeometry::bookHistograms(DQMStore::IBooker & ibooker, edm::Run const & Run, edm::EventSetup const & iSetup) {
+
+  if ( !detailPlot_ ) return ; 
+  const GEMGeometry* GEMGeometry_;
+
+  try{
+    edm::ESHandle<GEMGeometry> hGeom;
+    iSetup.get<MuonGeometryRecord>().get(hGeom);
+    GEMGeometry_=&*hGeom;
+  }
+  catch(edm::eventsetup::NoProxyException<GEMGeometry>& e){
+    edm::LogError("MuonGEMGeometry") << "+++ Error : GEM geometry  is unavailable on event loop. +++\n";
+    return;
+  }
+  
+  for( auto region : GEMGeometry_->regions()) {
+        TString title = TString::Format("Geometry's phi distribution on Region %d ; #phi(degree); ;",region->region());
+        TString name  = TString::Format("geo_phi_r%d",region->region());
+        auto temp_me = ibooker.book2D(name.Data(), title.Data(), 360000, -180., 180, 12,1,13);
+        temp_me->setBinLabel(1,"St1,La1_even",2);
+        temp_me->setBinLabel(2,"St1,La1_odd",2);
+        temp_me->setBinLabel(3,"St1,La2_even",2);
+        temp_me->setBinLabel(4,"St1,La2_odd",2);
+        temp_me->setBinLabel(5,"St2,La1_even",2);
+        temp_me->setBinLabel(6,"St2,La1_odd",2);
+        temp_me->setBinLabel(7,"St2,La2_even",2);
+        temp_me->setBinLabel(8,"St2,La2_odd",2);
+        temp_me->setBinLabel(9,"St3,La1_even",2);
+        temp_me->setBinLabel(10,"St3,La1_odd",2);
+        temp_me->setBinLabel(11,"St3,La2_even",2);
+        temp_me->setBinLabel(12,"St3,La2_odd",2);
+        theStdPlots.insert( std::map< UInt_t, MonitorElement*>::value_type(name.Hash(), temp_me ));
```

Comment:
Hello, rovere. I used "hash" function to get unique value from "string". Frankly speaking, I have a some trouble due to "const char*" comparision problem. So, I used Hash function to solve temporary. In addition, I heared unorderd_map is slower than map for small pairs. I will fix to use "map" with "std::string". Thanks for interest.